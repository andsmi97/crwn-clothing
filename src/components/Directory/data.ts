import { IMenuItem } from "../../types";
export const data: IMenuItem[] = [
  {
    title: "hats",
    imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
    id: 1,
    linkUrl: "shop/hats",
    size: "small"
  },
  {
    title: "jackets",
    imageUrl: "https://i.ibb.co/px2tCc3/jackets.png",
    id: 2,
    linkUrl: "shop/jackets",
    size: "small"
  },
  {
    title: "sneakers",
    imageUrl: "https://i.ibb.co/0jqHpnp/sneakers.png",
    id: 3,
    linkUrl: "shop/sneakers",
    size: "small"
  },
  {
    title: "womens",
    imageUrl: "https://i.ibb.co/GCCdy8t/womens.png",
    size: "large",
    id: 4,
    linkUrl: "shop/womens"
  },
  {
    title: "mens",
    imageUrl: "https://i.ibb.co/R70vBrQ/men.png",
    size: "large",
    id: 5,
    linkUrl: "shop/mens"
  }
];
