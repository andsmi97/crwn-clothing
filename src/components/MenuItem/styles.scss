.menu-item {
  min-width: 30%;
  height: 240px;
  flex: 1 1 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
  margin: 0 7.5px 15px;
  border: 2px solid black;
  transition: all 6s cubic-bezier(0.25, 0.45, 0.45, 0.95);
  position: relative;
  overflow: hidden;
  cursor: pointer;
  // transform: scale(1.05, 1.05);
  &.large {
    height: 380px;
  }

  &:first-child {
    margin-right: 7.5px;
  }

  &:last-child {
    margin-left: 7.5px;
  }
  &:hover {
    .content {
      background-color: rgba(255, 255, 255, 0.9);
    }
    .background-image {
      transform: scale(1.1, 1.1);
    }
  }
  .background-image {
    width: 100%;
    height: 100%;
    background-position: center;
    background-size: cover;
    position: absolute;
    transform: scale(1, 1);
    transition: all 6s cubic-bezier(0.25, 0.45, 0.45, 0.95);
    z-index: 0;
  }
  .content {
    height: 90px;
    padding: 0 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border: 1px solid black;
    background-color: rgba(255, 255, 255, 0.7);
    transition: all 0.5s;
    z-index: 1;
    .title {
      font-weight: bold;
      margin-bottom: 6px;
      font-size: 22px;
      color: #4a4a4a;
      margin: 0;
      text-transform: uppercase;
    }

    .subtitle {
      font-weight: lighter;
      font-size: 16px;
    }
  }
}
