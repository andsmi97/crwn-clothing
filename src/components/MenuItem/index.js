import React from "react";
import "./styles.scss";

export default ({ title, img, size }) => {
  return (
    <div className={`${size ? size : ""} menu-item`}>
      <div
        style={{
          backgroundImage: `url(${img})`
        }}
        className="background-image"
      />
      <div className="content">
        <h1 className="title">{title}</h1>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  );
};
