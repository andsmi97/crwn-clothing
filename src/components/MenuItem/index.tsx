import React from "react";
import "./styles.scss";
import { IMenuItem } from "../../types";
export default ({ title, imageUrl, size }: IMenuItem): JSX.Element => {
  return (
    <div className={`${size ? size : ""} menu-item`}>
      <div
        style={{
          backgroundImage: `url(${imageUrl})`
        }}
        className="background-image"
      />
      <div className="content">
        <h1 className="title">{title}</h1>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  );
};
