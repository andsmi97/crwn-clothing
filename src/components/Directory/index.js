import React, { useState } from "react";
import { data } from "./data";
import MenuItem from "../MenuItem";
import "./styles.scss";

export default () => {
  const [selection, setSelection] = useState(data);
  return (
    <div className="directory-menu">
      {selection.map(({ title, id, imageUrl, linkUrl, size }) => (
        <MenuItem
          title={title}
          key={id}
          img={imageUrl}
          link={linkUrl}
          size={size}
        />
      ))}
    </div>
  );
};
