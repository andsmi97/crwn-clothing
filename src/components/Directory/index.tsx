import React, { useState } from "react";
import { data } from "./data";
import MenuItem from "../MenuItem";
import { IMenuItem } from "../../types";
import "./styles.scss";

export default (): JSX.Element => {
  const [selection] = useState<IMenuItem[]>(data);
  return (
    <div className="directory-menu">
      {selection.map(({ id, ...restProps }) => (
        <MenuItem key={id} id={id} {...restProps} />
      ))}
    </div>
  );
};
