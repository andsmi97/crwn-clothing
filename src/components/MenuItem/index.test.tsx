import React from "react";
import { shallow } from "enzyme";
import MenuItem from "../MenuItem";
import { data } from "../Directory/data";
describe("<MenuItem />", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(
      <MenuItem
        id={data[0].id}
        title={data[0].title}
        imageUrl={data[0].imageUrl}
        size={data[0].size}
        linkUrl={data[0].linkUrl}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
