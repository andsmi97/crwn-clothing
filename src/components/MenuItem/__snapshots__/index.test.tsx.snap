// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> matches snapshot 1`] = `ShallowWrapper {}`;
